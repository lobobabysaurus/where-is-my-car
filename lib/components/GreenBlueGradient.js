import { LinearGradient } from 'expo'
import PropTypes from 'prop-types'
import React from 'react'

const GreenBlueGradient = ({children, style}) => {
  const startColor = '#50C878'
  const endColor = '#00008B'
  const gradient = [startColor, endColor]
  return <LinearGradient style={style} colors={gradient}>
    {children}
  </LinearGradient>
}

GreenBlueGradient.propTypes = {
  children: PropTypes.node,
  style: PropTypes.object,
}

export default GreenBlueGradient
