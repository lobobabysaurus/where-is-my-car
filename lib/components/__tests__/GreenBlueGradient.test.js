import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'

import GreenBlueGradient from '../GreenBlueGradient'

describe('GreenBlueGradient', () => {
  test('renders correctly', () => {
    const component = renderer.create(
      <GreenBlueGradient style={{maxWidth: 23}}>
        <Text>Im in a gradient</Text>
      </GreenBlueGradient>
    ).toJSON()
    expect(component).toMatchSnapshot()
  })
})
