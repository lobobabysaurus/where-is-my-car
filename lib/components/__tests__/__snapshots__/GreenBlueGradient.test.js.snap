// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreenBlueGradient renders correctly 1`] = `
<ExponentLinearGradient
  colors={
    Array [
      4283484280,
      4278190219,
    ]
  }
  style={
    Object {
      "maxWidth": 23,
    }
  }
>
  <Text>
    Im in a gradient
  </Text>
</ExponentLinearGradient>
`;
